journal.py

```diff
@@ -0,0 +1,199 @@
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    body_text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow)
+
+    @classmethod
+    def write(cls, title=None, body_text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, body_text=body_text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+    @classmethod
+    def get_article(cls, article_id, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter(cls.id == article_id).all()
+
+
+def init_db():
+    engine = sa.create_engine(DATABASE_URL, echo=False)
+    Base.metadata.create_all(engine)
+
+
+def do_login(request):
+    username = request.params.get('username', None)
+    password = request.params.get('password', None)
+    if not (username and password):
+        raise ValueError('both username and password are required')
+
+    settings = request.registry.settings
+    manager = BCRYPTPasswordManager()
+    if username == settings.get('auth.username', ''):
+        hashed = settings.get('auth.password', '')
+        return manager.check(hashed, password)
+    return False
+
+
+def main():
+    """Create a configured wsgi app"""
+    settings = {}
+    debug = os.environ.get('DEBUG', True)
+    settings['reload_all'] = debug
+    settings['debug_all'] = debug
+    #  Adding these as step3
+    settings['auth.username'] = os.environ.get('AUTH_USERNAME', 'admin')
+    manager = BCRYPTPasswordManager()
+    settings['auth.password'] = os.environ.get(
+        'AUTH_PASSWORD', manager.encode('secret')
+        )
+    if not os.environ.get('TESTING', False):
+        #  Connect to database only if not in testing
+        engine = sa.create_engine(DATABASE_URL)
+        DBSession.configure(bind=engine)
+    # add a "secret" value for auth tkt signing
+    auth_secret = os.environ.get('JOURNAL_AUTH_SECRET', 'itsaseekrit')
+
+    # configuration setup
+    config = Configurator(
+        settings=settings,
+        authentication_policy=AuthTktAuthenticationPolicy(
+            secret=auth_secret,
+            hashalg='sha512'
+            ),
+        authorization_policy=ACLAuthorizationPolicy(),
+    )
+    # Allow packages to declare their configurations
+    config.include('pyramid_tm')
+    config.include('pyramid_jinja2')
+    config.add_route('home', '/')
+    config.add_route('add', '/add')
+    config.add_route('login', '/login')
+    config.add_route('logout', '/logout')
+    config.add_route('detail', '/detail/{id}')
+    config.add_route('new', '/new')
+    config.add_route('edit', '/edit/{id}')
```

Comment:
am I missing the view where you actually set new values on an existing entry?  This route appears to match to a view that returns the specified entry data for display in a form, but I'm not seeing the second part, where that form is submitted and the existing entry is updated.